modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCDataStoreOnlineTest.java

```diff
@@ -139,16 +139,19 @@ public void testCreateSchemaWithConstraints() throws Exception {
         //assertEquals(ft2, featureType);
 
         //grab a writer
-        FeatureWriter w = dataStore.getFeatureWriter( tname("ft2"),Transaction.AUTO_COMMIT);
+        FeatureWriter<SimpleFeatureType, SimpleFeature> w =
+                dataStore.getFeatureWriter( tname("ft2"),Transaction.AUTO_COMMIT);
         w.hasNext();
 
-        SimpleFeature f = (SimpleFeature) w.next();
+        SimpleFeature f = w.next();
         f.setAttribute( 1, new Integer(0));
         f.setAttribute( 2, "hello");
         w.write();
+        w.close();
```

Comment:
Ok, this is the type of backwards compatibilty change that scares me. Due to batching, the writers have to be closed before anything else in the same transaction can see the modifications, which for some applications will be a serious problem (since they might need to write a little, read a little, write some more), especially since it will be hard to locate for those upgrading (and people notably do not read upgrade notes until they figure they are having problems).
This may require a discussion in a larger group, but we might want to have batching activated by some flag, and off by default.